synapse/storage/databases/main/devices.py

```diff
@@ -120,6 +122,26 @@ async def get_device(self, user_id: str, device_id: str) -> Dict[str, Any]:
             desc="get_device",
         )
 
+    async def get_device_opt(
```

Comment:
I think it'd be nicer if we just switched `get_device` to return an `Optional`, and then change up the callers to look for `None` instead of a `StoreError`.

That would also fix up callers that seem to currently 500 if a device isn't found(!)

Reply:
Span out to #11565 .